### DISCUSSION TITLE
How implement JWT in .net c#?

### DISCUSSION THREAD

**Discussion by aisakk** [dotnet/aspnetcore#53592] [Q&A]
Hello, I am trying to create a minimals web API with .net or controller api, and I want to implement JWT, I am guided by the microsoft documentation but I get dotnet users-jwt is only for testing and I would like to implement JwtSecurity from the package System.IdentityModel.Tokens.Jwt or Microsoft.IdentityModel.Tokens.Jwt but when I go to the documentation there is no example of how to implement it, I would like to know if they are on it? or if there is an example of how to do it?.

The only reference I see from microsoft is this https://learn.microsoft.com/es-es/aspnet/core/security/authentication/jwt-authn?view=aspnetcore-8.0&tabs=linux but it seems dotnet users-jwt is only for testing and I would like to generate the token myself and see if there is a more detailed specification of JwtSecurityToken ([System.IdentityModel.Tokens.Jwt] and (Microsoft.IdentityModel.Jwt)

**Comment by martincostello** [comment]
Take a look at this blog post which will help point you in the right direction about how you go about implementing this complicated topic: [_Bearer Token Authentication in ASP.NET Core_](https://devblogs.microsoft.com/dotnet/bearer-token-authentication-in-asp-net-core/).

Implementing authentication and authorization correctly and securely is a broad topic.
